### PR TITLE
feat(Connector):[Adyen]Implement ACH Direct Debits for Adyen

### PR DESCRIPTION
**Pull request by swangi-kumari**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Implemented Bank Debit Data for Adyen Payment Method.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
In Adyen connector ACH Bank Debits was not implemented, but it is important payment method which is needed in market so I implemented it.

## How did you test it?
I tested it using Postman.
Screenshot 2023-05-03 at 11 45 42 AM

Screenshot 2023-05-03 at 11 46 07 AM

Screenshot 2023-05-03 at 11 46 00 AM

Screenshot 2023-05-03 at 11 46 18 AM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/adyen/transformers.rs [hunk 1]
```diff
@@ -1098,7 +1179,7 @@ impl<'a>
             shopper_locale,
             billing_address: None,
             delivery_address: None,
-            country_code: country,
+            country_code: None,
```
Comment (ArjunKarthik) [inline crates/router/src/connector/adyen/transformers.rs:1190]: Why we are making country as None here?

File: crates/router/src/connector/adyen/transformers.rs [hunk 2]
```diff
@@ -1057,6 +1100,44 @@ impl<'a> TryFrom<(&types::PaymentsAuthorizeRouterData, &api::Card)> for AdyenPay
...
+            shopper_name: None,
+            shopper_locale:None,
+            shopper_email: item.request.email.clone(),
+            telephone_number: None,
+            billing_address: None,
+            delivery_address: None,
+            country_code: None,
+            line_items: None,
+        };
+        crate::logger::debug!(adyen_request=?request);
```
Comment (ArjunKarthik) [inline crates/router/src/connector/adyen/transformers.rs:1144]: remove logger. We should not log anything inside connector code

File: crates/router/src/connector/adyen/transformers.rs [hunk 3]
```diff
@@ -1057,6 +1100,44 @@ impl<'a> TryFrom<(&types::PaymentsAuthorizeRouterData, &api::Card)> for AdyenPay
...
+
+    fn try_from(value: (&types::PaymentsAuthorizeRouterData, &api_models::payments::BankDebitData)
+    ) -> Result<Self, Self::Error> {
+        let (item, bank_debit_data) = value;
+        let amount = get_amount_data(item);
+        let auth_type = AdyenAuthType::try_from(&item.connector_auth_type)?;
+        let shopper_interaction = AdyenShopperInteraction::from(item);
+        let recurring_processing_model = get_recurring_processing_model(item);
+        let browser_info = get_browser_info(item);
+        let additional_data = get_additional_data(item);
```
Comment (ArjunKarthik) [inline crates/router/src/connector/adyen/transformers.rs:1122]: Does Adyen support manual capture for bank debits? Cause `get_additional_data` will add authorisation_type type in the request?

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

File: crates/api_models/src/payments.rs [hunk 4]
```diff
@@ -475,6 +475,9 @@ pub enum BankDebitData {
         /// Routing number for ach bank debit payment
         #[schema(value_type = String, example = "110000000")]
         routing_number: Secret<String>,
+
+        #[schema(value_type = String, example = "John Test")]
+        bank_account_holder_name: Secret<String>,
```
Comment (jagan-jaya) [inline crates/api_models/src/payments.rs:480]: is this mandatory for all connector? stripe was implemented without this field right?

File: crates/router/src/connector/adyen/transformers.rs (hunk 1 above)
Comment (ArjunKarthik) [reply crates/router/src/connector/adyen/transformers.rs:1342]: This was resolved without any comments/changes?

File: crates/router/src/connector/adyen/transformers.rs [hunk 5]
```diff
@@ -1,5 +1,7 @@
 use api_models::{
-    enums::DisputeStage, payments::MandateReferenceId, webhooks::IncomingWebhookEvent,
+    enums::DisputeStage,
+    payments::{self, MandateReferenceId},
+    webhooks::IncomingWebhookEvent,
```
Comment (ArjunKarthik) [inline crates/router/src/connector/adyen/transformers.rs:4]:
```suggestion
use api_models::{enums, payments, webhooks};
```

Lets change this, Specificc import can cause conflicts in the import part, So lets keep it in this way

**Review by ArjunKarthik — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by jagan-jaya — APPROVED** [review]

**Review by jarnura — DISMISSED** [review]

**Review by jagan-jaya — APPROVED** [review]

**Pull request merged**